File: testdata/java/src/main/java/com/example/services/PaymentService.java
package com.example.services;

import java.util.List;
import java.util.Optional;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;
import com.example.domain.Payment;

@Service
public class PaymentService {

    private final String apiKey;
    private final RestTemplate restTemplate;

    public PaymentService(String apiKey, RestTemplate restTemplate) {
        this.apiKey = apiKey;
        this.restTemplate = restTemplate;
    }

    public Payment charge(int amount, String token) {
        if (amount <= 0) {
            throw new IllegalArgumentException("Amount must be positive");
        }
        return new Payment(amount, token);
    }

    public boolean refund(String chargeId) {
        if (chargeId == null || chargeId.isEmpty()) {
            return false;
        }
        return true;
    }

    private void internalHelper() {}
}

File: testdata/csharp/src/Services/PaymentService.cs
using System;
using System.Threading.Tasks;
using Stripe;
using Microsoft.Extensions.Logging;

namespace MyApp.Services;

public class PaymentService
{
    private readonly ILogger<PaymentService> _logger;
    private readonly string _apiKey;

    public PaymentService(ILogger<PaymentService> logger, string apiKey)
    {
        _logger = logger;
        _apiKey = apiKey;
    }

    public async Task<string> ChargeAsync(int amount, string token)
    {
        if (amount <= 0)
            throw new ArgumentException("Amount must be positive", nameof(amount));

        _logger.LogInformation("Charging {Amount}", amount);
        return Guid.NewGuid().ToString();
    }

    public bool Refund(string chargeId)
    {
        if (string.IsNullOrEmpty(chargeId))
            return false;

        return true;
    }

    private void InternalHelper() { }
}
